Indexer will throw an Error if temporalCoverage aspects intervals is an empty array

### Problem description

Indexer will throw an Error if temporalCoverage aspects intervals is an empty array:
```
[ERROR] [04/26/2019 02:31:34.652] [indexer-akka.actor.default-dispatcher-4] [akka.actor.ActorSystemImpl(indexer)] Exception encountered
java.lang.RuntimeException: java.lang.IndexOutOfBoundsException: Too little elements in array: [] size: 0 index: 0
	at spray.json.lenses.package$GetOrThrow.getOrThrow(package.scala:26)
	at spray.json.lenses.LensImpl.get(Lens.scala:33)
	at spray.json.lenses.ExtraImplicits$RichJsValue$class.extract(ExtraImplicits.scala:19)
	at spray.json.lenses.ExtraImplicits$$anon$1.extract(ExtraImplicits.scala:28)
	at au.csiro.data61.magda.model.Registry$RegistryConverters$class.convertRegistryDataSet(Registry.scala:273)
	at au.csiro.data61.magda.indexer.external.registry.WebhookApi.convertRegistryDataSet(WebhookApi.scala:21)
	at au.csiro.data61.magda.indexer.external.registry.WebhookApi$$anonfun$1$$anonfun$apply$1$$anonfun$7$$anonfun$8.apply(WebhookApi.scala:65)
	at au.csiro.data61.magda.indexer.external.registry.WebhookApi$$anonfun$1$$anonfun$apply$1$$anonfun$7$$anonfun$8.apply(WebhookApi.scala:64)
	at scala.collection.immutable.List.map(List.scala:288)
```

The problem is around this line:

https://github.com/magda-io/magda/blob/c8cdfab7bf928143328a1eafcd36915b835cd584/magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala#L253

`element(0)` will trigger an error for an empty array

### Problem reproduction steps

- Create a new dataset (using new dataset UI) without specifying temporalCoverage
- Wait for indexer starts to index this dataset

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -4,6 +4,10 @@ Search:
 
 -   Prevent freeText query from being None which will cause score to be 0
 
+Indexer:
+
+-   Fixed indexer throws an error when temporalCoverage aspects intervals is an empty array
+
 Others:
 
 -   Made registry-api DB pool settings configurable via Helm

--- a/magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala
+++ b/magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala
@@ -250,8 +250,26 @@ object Registry {
         case _ => 1d
       }
 
-      val coverageStart = ApiDate(tryParseDate(temporalCoverage.extract[String]('intervals.? / element(0) / 'start.?)), dcatStrings.extract[String]('temporal.? / 'start.?).getOrElse(""))
-      val coverageEnd = ApiDate(tryParseDate(temporalCoverage.extract[String]('intervals.? / element(0) / 'end.?)), dcatStrings.extract[String]('temporal.? / 'end.?).getOrElse(""))
+      // --- intervals could be an empty array
+      // --- put in a Try to avoid exceptions
+      val coverageStart = ApiDate(tryParseDate(
+        Try[Option[String]]{
+          temporalCoverage.extract[String]('intervals.? / element(0) / 'start.?)
+        } match {
+          case Success(Some(v)) => Some(v)
+          case _ => None
+        }
+      ), dcatStrings.extract[String]('temporal.? / 'start.?).getOrElse(""))
+
+      val coverageEnd = ApiDate(tryParseDate(
+        Try[Option[String]]{
+          temporalCoverage.extract[String]('intervals.? / element(0) / 'end.?)
+        } match {
+          case Success(Some(v)) => Some(v)
+          case _ => None
+        }
+      ), dcatStrings.extract[String]('temporal.? / 'end.?).getOrElse(""))
+
       val temporal = (coverageStart, coverageEnd) match {
         case (ApiDate(None, ""), ApiDate(None, "")) => None
         case (ApiDate(None, ""), end)               => Some(PeriodOfTime(None, Some(end)))

